### DISCUSSION TITLE
格式和配色可以分开，那样便于维护。

### DISCUSSION THREAD

**Discussion by voncin** [liangjingkanji/DrakeTyporaTheme#52] [Ideas]
一个系列的主题，共性都可以拆出来。排版格式，代码块配色，变量，都可以拆。拆出来以后，就不用挨个改了。开发者一劳永逸，用户还可以DIY。

**Comment by liangjingkanji** [comment]
考虑了下, 因为没有说某个主题就只能修改什么地方. 所以我还是不建议抽出公用(因为样式可能还会存在冲突无法覆盖). 后续也可能会有更多样式区别更明显主题出现

**Reply by voncin** [reply]
拆了的话比较整齐, 放在一起会比较乱. 

重新写只是感觉麻烦, 其实差不多.
非常麻烦一次, 和每次麻烦一下的区别.

**Reply by liangjingkanji** [reply]
所有主题继承一种样式. 那如果某个主题需要修改你抽出的属性怎么办. 可能可以覆盖, 但是就不好处理优先级关系.

因为所有主题并不是简单的换颜色, 所以不考虑抽出. 而且抽出以后更乱. 目前每个模块样式都在一块, 一旦抽出一个模块的样式就会在两个文件中更乱

我现在同时维护10个主题并没有觉得多麻烦. 只是一键替换而已

而且每个主题的全局颜色值也做不到给基础样式使用, 一旦拆开就会很麻烦, 如果你觉得我的想法有问题. 可以自己尝试拆分提交PR给我看

**Reply by voncin** [reply]
都可以的，我的意见不太成熟。您酌情采纳。

**Reply by liangjingkanji** [reply]
反正在我的认知是会更麻烦

**Comment by liangjingkanji** [comment]
但是你如果想全部修改所有主题可以使用全局替换(我也是一直如此维护). 因为样式位置基本都是一样的

**Comment by huangjingling** [comment]
你好，我想问一下那个drake-google的主题里面的==会自动合并成一个大的=，怎么让他们不合并呢？

**Reply by liangjingkanji** [reply]
这是默认字体JetBrains导致的, 你替换成其他字体. 或者使用JetBrainsMono不带`连字`特性的版本

**Reply by huangjingling** [reply]
能请教一下具体的修改方法吗，我不是很了解这个语言规则

**Reply by huangjingling** [reply]
在font.css里面修改？

**Reply by liangjingkanji** [reply]
你稍等下我出个文档, 因为有些常见问题反复有人问我收集下

**Reply by huangjingling** [reply]
嗯，好的，谢谢

**Reply by liangjingkanji** [reply]
你去看[常见问题](https://github.com/liangjingkanji/DrakeTyporaTheme/blob/master/issues.md)
